### DISCUSSION TITLE
`chezmoi dump` and absolute paths

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [twpayne/chezmoi#2627] [Q&A]
I noticed that when I run `chezmoi dump --include scripts` I get the script name without any decoration/flag.

for example:
```
{
  ".chezmoiscripts/my_script_name.sh": {
    "type": "script",
    "name": ".chezmoiscripts/my_script_name.sh",
    "contents": "#! /usr/bin/env bash\n\n...",
    "condition": "always"
  }
}
```

But if I want to perform an action on this script, I need to find the script within the `chemoi` directory on my own, not to mention other than the fact that this is a script, I don't have any indication of the absolute path of the file.

Is there a way for dump to include the full path?

Why I'm asking?
I'm writing an edit script with auto complete.
The purpose of the script is so that after editing, it will prompt the user whether to add the files to chezmoi state or not. if it's not inside the `.chezmoi` dir, no prompt is needed.  The prompt side works well, but for the auto-complete to list the files, I want easy access to the scripts as well.

`dump` seems the safest (and fastest) way to get the list of what is already being managed by `chezmoi`, but no absolute path, which makes it a bit of a chore for me.

**Comment by twpayne** [comment]
I think you want [`chezmoi managed --path-style=absolute`](https://www.chezmoi.io/reference/commands/managed/).

You can also use [`chezmoi source-path`](https://www.chezmoi.io/reference/commands/source-path/).

Note also that the key (`.chezmoiscripts/my_script_name.sh` in your example) is the path relative to chezmoi's root directory.

**Reply by Lockszmith-GH** [reply]
Thanks.
Yes, I know the chezmoiscript path is in the `chezmoi source-path`, I've resolved to querying scripts and templates separately because of that.

However, if a file has `private` or `executable` or `run_before` in the name, chezmoi doesn't report that.
<br/>For now, I managed to deal with it, in a clunky way, but I would really like to get this information directly from chezmoi if possible.

I think what I want is indeed a `chezmoi managed --path-style=absolute`, but I get the following:
> ```
> chezmoi managed --path-style=absolute
> chezmoi: unknown flag: --path-style

Note that the documentation page you pointed to does not have that flag either.

Should I open an FR?

<details>
<summary>chezmoi doctor output</summary>

```
RESULT    CHECK                MESSAGE
ok        version              v2.27.3, commit feab4e788831e803d84a544739dfb8188245deae, built at 2022-12-05T00:31:21Z, built by goreleaser
ok        latest-version       v2.27.3
ok        os-arch              linux/amd64 (Pop!_OS 22.04 LTS)
ok        uname                Linux pop-os 6.0.6-76060006-generic #202210290932~1669062050~22.04~d94609a SMP PREEMPT_DYNAMIC Mon N x86_64 x86_64 x86_64 GNU/Linux
ok        go-version           go1.19.3 (gc)
ok        executable           ~/bin/chezmoi
ok        upgrade-method       replace-executable
ok        config-file          ~/.config/chezmoi/chezmoi.toml, last modified 2022-12-08T08:58:32-05:00
warning   source-dir           ~/.local/share/chezmoi is a git working tree (dirty)
ok        suspicious-entries   no suspicious entries
warning   working-tree         ~/.local/share/chezmoi is a git working tree (dirty)
ok        dest-dir             ~ is a directory
ok        umask                002
ok        cd-command           found /bin/bash
ok        cd-args              /bin/bash
ok        diff-command         found ~/.local/bin/icdiff
ok        edit-command         found /usr/bin/vi
ok        edit-args            /usr/bin/vi
ok        git-command          found /usr/bin/git, version 2.34.1
ok        merge-command        found /usr/bin/vimdiff
ok        shell-command        found /bin/bash
ok        shell-args           /bin/bash
ok        age-command          found /usr/bin/age, version 1.0.0
ok        gpg-command          found /usr/bin/gpg, version 2.2.27
info      pinentry-command     not set
info      1password-command    op not found in $PATH
info      bitwarden-command    bw not found in $PATH
info      gopass-command       gopass not found in $PATH
info      keepassxc-command    keepassxc-cli not found in $PATH
info      keepassxc-db         not set
info      keeper-command       keeper not found in $PATH
info      lastpass-command     lpass not found in $PATH
ok        pass-command         found /usr/bin/pass, version 1.7.4
info      passhole-command     ph not found in $PATH
info      vault-command        vault not found in $PATH
info      secret-command       not set
```
</details>

**Reply by Lockszmith-GH** [reply]
Oh, I [see](https://github.com/twpayne/chezmoi/blob/243aaa88dba687e536e679aed8a022defef3aa6e/pkg/cmd/managedcmd.go#L32), it's a brand new feature you coded in ~2 weeks ago, I guess it hadn't made it to a release yet.

This will be awesome.

**Reply by Lockszmith-GH** [reply]
And just now, saw that you pushed a new version 18 hours ago, and indeed `--path-style` is there, however, the paths are inaccurate (or at least, not what I need).

For example, I have 2 scripts in my git repo in a `.chezmoiscripts` directory: `run_after_apply_base_chezmoi.sh` and `run_once_after_10_cz_init`

Running `chezmoi managed --path-style absolute` I get:
```
~/.chezmoiscripts/10_cz_init
~/.chezmoiscripts/apply_base_chezmoi.sh
```
when I was expecting to see:
```
~/.local/share/chezmoi/.chezmoiscripts/run_after_apply_apply_base_chezmoi.sh
~/.local/share/chezmoi/.chezmoiscripts/run_once_after_10_cz_init
```

As `path-style` seems to be a new feature, any chance to see a change in implementation? maybe add an `actual` path-style?

**Reply by twpayne** [reply]
Ah, I see now. You want source paths, but `chezmoi managed` gives you target paths.

#2633 adds `--path-style=source-absolute` to `chezmoi managed`.
